discretisedfield/tests/test_field.py

```diff
@@ -1796,6 +1796,78 @@ def value_fun(point):
     assert np.allclose(f.diff("z").mean(), (0, 0, 0))
 
 
+def test_diff_valid():
```

Comment:
Please add a second example on a 1d mesh with floating-point values and a valid array with single-element sections, i.e. something like:

```suggestion
def test_diff_valid():
    # 1d mesh
    mesh = df.Mesh(p1=-5e-9, p2=10e-9, n=10)
    valid = [True, False, False, True, True, True, False, True, False, False]
    f = df.Field(mesh, nvdim=1, value=lambda p: (p[0] * 1e9)**2, valid=valid)
    
    # TODO add tests
```